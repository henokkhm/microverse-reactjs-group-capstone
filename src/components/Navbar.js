import { NavLink } from 'react-router-dom'; // If you're using React Router for navigation

import styles from '../styles/Navbar.module.css';

function Navbar() {
  return (
    <nav className={styles.navigation}>
      <ul className={styles.navList}>
        <li className={styles.navListItem}>
          <NavLink className={styles.navLink} to="/">
            Home
          </NavLink>
        </li>
        <li>
          <NavLink
            className={`${styles.navLink} ${styles.myProfileLink}`}
            to="/my-profile"
          >
            My Profile
          </NavLink>
        </li>
      </ul>
    </nav>
  );
}

export default Navbar;
