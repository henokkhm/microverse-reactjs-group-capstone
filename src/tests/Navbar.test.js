import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import Navbar from '../components/Navbar';

it('The Navbar component renders correctly', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
