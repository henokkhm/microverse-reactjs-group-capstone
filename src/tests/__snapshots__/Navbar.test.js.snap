// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Navbar component renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="navigation"
  >
    <ul
      class="navList"
    >
      <li
        class="navListItem"
      >
        <a
          aria-current="page"
          class="navLink active"
          data-testid="navlink-rockets-page"
          href="/"
        >
          Rockets
        </a>
      </li>
      <li
        class="navListItem"
      >
        <a
          class="navLink"
          data-testid="navlink-missions-page"
          href="/missions"
        >
          Missions
        </a>
      </li>
      <li
        class="navListItem"
      >
        <a
          class="navLink"
          data-testid="navlink-my-profile-page"
          href="/my-profile"
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
